core/src/main/scala/aima/core/environment/vacuum/VacuumEnvironment.scala

```diff
@@ -6,34 +6,14 @@ import aima.core.random.DefaultRandomness
 /**
   * @author Shawn Garner
   */
-case class VacuumEnvironment(map: VacuumMap = VacuumMap()) extends Environment[VacuumAction, VacuumPercept] { self =>
+case class Vacuum(map: VacuumMap = VacuumMap()) extends Environment[Vacuum, VacuumPercept, VacuumAction] {
```

Comment:
@benderpremier this is really strange that the Vacuum is an agent and an environment.  Can you separate these out as Vacuum as the agent and another class VacuumEnvironment as the environment?
I do agree with getting rid of the self type though.

Reply:
Sorry about that, yeah I agree with you I'll rename it back to `VacuumEnvironment`. I can't remember why I did that change.